subprojects/testing-native/src/integTest/groovy/org/gradle/nativeplatform/test/xctest/SwiftXCTestIntegrationTest.groovy

```diff
@@ -82,21 +87,26 @@ apply plugin: 'swift-executable'
         succeeds("test")
 
         then:
-        result.assertTasksSkipped(":compileTestSwift", ":linkTest", ":createXcTestBundle", ":xcTest", ":test")
+        result.assertTasksSkipped(":compileSwift", ":compileTestSwift", ":linkTest", ":createXcTestBundle", ":xcTest", ":test")
     }
 
-    // TODO - Integrate automatically with the Swift executable
-    def "applying the xctest plugin together with swift-executable will not assemble the executable by convention"() {
+    def "xctest plugin can test public and internal feature of a Swift executable"() {
         def app = new SwiftApp()
         def test = new SwiftXcTestTestApp([
             newTestSuite("FooTestSuite", [
-                newTestCase("testFoo", TestElement.TestCase.Result.PASS)
-            ])
+                newTestCase("testInternalMultiply", TestElement.TestCase.Result.PASS,
+                    "XCTAssert(multiply(a: 21, b: 2) == ${app.multiply.multiply(21, 2)})"),
+                newTestCase("testPublicSum", TestElement.TestCase.Result.PASS,
+                    "XCTAssert(sum(a: 40, b: 2) == ${app.sum.sum(40, 2)})"),
+            ], ["App"])
         ])
 
         given:
+        settingsFile << "rootProject.name = 'App'"
         buildFile << """
 apply plugin: 'swift-executable'
+
+linkTest.source = project.files(new HashSet(linkTest.source.from)).filter { !it.name.equals("main.o") }
```

Comment:
My bad for not being more precise. You are right, that including `main.o` result in linking errors. To solve this, we would need to detect the symbol and find a way to ignore it. One idea that I never had a chance to try was renaming the `_main` symbol to something else to avoid linking errors. However, in the presence of multiple `_main` symbols, we don't want to rename the principal one.